File: backend/src/main/java/space/davids_digital/sweetie/orm/repository/UserSessionRepository.java
package space.davids_digital.sweetie.orm.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import space.davids_digital.sweetie.orm.entity.UserSessionEntity;

import java.time.ZonedDateTime;
import java.util.Collection;
import java.util.UUID;

@Repository
public interface UserSessionRepository extends CrudRepository<UserSessionEntity, UUID> {
    Collection<UserSessionEntity> findAllByValidUntilBefore(ZonedDateTime validUntil);
    Collection<UserSessionEntity> findAllByUserVkIdAndValidUntilAfter(long userVkId, ZonedDateTime validUntil);
    void deleteByValidUntilBefore(ZonedDateTime validUntil);
}

File: backend/src/main/java/space/davids_digital/sweetie/orm/repository/UserSessionRepository.kt
package space.davids_digital.sweetie.orm.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import space.davids_digital.sweetie.orm.entity.UserSessionEntity
import java.time.ZonedDateTime
import java.util.*

@Repository
interface UserSessionRepository: CrudRepository<UserSessionEntity, UUID> {
    fun findAllByValidUntilBefore(validUntil: ZonedDateTime): Collection<UserSessionEntity>
    fun findAllByUserVkIdAndValidUntilAfter(userVkId: Long, validUntil: ZonedDateTime): Collection<UserSessionEntity>
    fun deleteByValidUntilBefore(validUntil: ZonedDateTime)
}
